core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -162,6 +161,25 @@ public ClusterState execute(ClusterState currentState) throws Exception {
         });
     }
 
+    void validatePipeline(Map<DiscoveryNode, IngestInfo> ingestInfos, PutPipelineRequest request) throws Exception {
```

Comment:
> That said you can store a pipeline, the stop some node, uninstall some plugin and start the node, but the pipeline is still there and not guaranteed to work?

yes, it is difficult to protect against that. There isn't a hook point that prevents ingest from starting the node if there is a pipeline with unsupported processors. Even if that exists then there is always the case that the local cluster state is outdated and upon joining the cluster new cluster state contain pipelines with unsupported processors. Also node startup isn't the only scenario when this is a problem, also when a node gets isolated from the cluster and then rejoins and in the meantime a pipeline has been added with an unsupported processor.

What happens now in that case is that pipelines with unsupported processors are loaded on these nodes and thus can't be used there. (index/bulk requests with ingest fail). A good way to verify if all nodes were able to load a pipeline is to check the node stats api and check if the new ingest stats for that pipeline have been set (#16915).